requests/structures.py

```diff
@@ -103,3 +104,94 @@ def __getitem__(self, key):
 
     def get(self, key, default=None):
         return self.__dict__.get(key, default)
+
+
+class TimedCache(object):
+    """
+    Evicts entries after expiration_secs.
+    If none are expired and maxlen is hit,
+    will evict the oldest cached entry
+    """
+
+    def __init__(self, maxlen=32, expiration_secs=60):
+        """
+        :param maxlen: most number of entries to hold on to
+        :param expiration_secs: the number of seconds to hold on
+        to entries
+        """
+
+        self.maxlen = maxlen
+        self.expiration_secs = expiration_secs
+        self._dict = dict()
+
+    def __repr__(self):
+        return '<timed-cache maxlen:%d len:%d>' % \
+            (self.maxlen, len(self._dict))
+
+    def __getitem__(self, key):
+        """
+        Look up an item in the cache. If the item
+        has already expired, it will be invalidated and not returned
+        :param key: which entry to look up
+        :return: the value in the cache, or None
+        """
+
+        return self.get(key, None)
+
+    def get(self, key, default=None):
+        """
+        Look up an item in the cache. If the item
+        has already expired, it will be invalidated and not returned
+        :param key: which entry to look up
+        :return: the value in the cache, or None
+        """
+
+        found = self._dict.get(key, default)
+        if not found:
+            return default
+
+        occurred, value = found
+        now = int(time.time())
+
+        if now - occurred > self.expiration_secs:
+            del self._dict[key]
+            return default
+        else:
+            return value
+
+    def __setitem__(self, key, value):
+        """
+        Locates the value at lookup key,
+        if cache is full, will evict the first
+        found expired entry, otherwise will evict
+        the oldest non-expired item
```

Comment:
You're wrapping this paragraph *very* short. Feel free to go all the way out to 89 columns!